x-pack/test/cases_api_integration/security_and_spaces/tests/common/comments/delete_comment.ts

```diff
@@ -91,9 +111,299 @@ export default ({ getService }: FtrProviderContext): void => {
       });
     });
 
-    describe('rbac', () => {
-      const supertestWithoutAuth = getService('supertestWithoutAuth');
+    describe('alerts', () => {
+      describe('security_solution', () => {
+        const createCaseAttachAlertAndDeleteAlert = async ({
+          totalCases,
+          indexOfCaseToDelete,
+          expectedHttpCode = 204,
+          auth = { user: superUser, space: 'space1' },
+        }: {
+          totalCases: number;
+          indexOfCaseToDelete: number;
+          expectedHttpCode?: number;
+          auth?: { user: User; space: string | null };
+        }) => {
+          const cases = await Promise.all(
+            [...Array(totalCases).keys()].map((index) =>
+              createCase(
+                supertestWithoutAuth,
+                {
+                  ...postCaseReq,
+                  settings: { syncAlerts: false },
+                },
+                200,
+                { user: superUser, space: 'space1' }
+              )
+            )
+          );
+
+          const signals = await createSecuritySolutionAlerts(supertest, log);
+          const alerts = [signals.hits.hits[0], signals.hits.hits[1]];
+          const updatedCases = [];
+
+          for (const theCase of cases) {
+            const updatedCase = await createComment({
```

Comment:
Could we do a `Promise.all` here?

Reply:
The `Promise.all` produces some weird race conditions/conflicts and the alerts are not updated correctly with the case ids.